internal/provider/resource_application.go

```diff
@@ -147,6 +149,14 @@ func (r *applicationResource) Schema(_ context.Context, _ resource.SchemaRequest
 					stringplanmodifier.RequiresReplaceIfConfigured(),
 				},
 			},
+			"model_type": schema.StringAttribute{
```

Comment:
issue: this should be handled in the internal/juju code and not at the user level. The info is available in the modelUUIDcache.

Reply:
Then the check should move to the Update method, where the model type is available.